mapping_interface.py

```diff
@@ -0,0 +1,114 @@
+from cartesian_coordinate import CartesianCoordinate
+from enum import Enum
+
+
+class PointState(Enum):
+    """
+    @brief PointState enum used to determine what the state of a location is.
+    """
+
+    unoccupied = 0, "unoccupied"
+    occupied = 1, "occupied"
+
+    def __new__(cls, value, name):
+        """
+        @brief PointState enum used to determine what the state of a location is.
+        @details 
+        This method creates a new instance of the PointState class
+        and sets the value and name.
+        @param value The value of the PointState you want to set to the new PointState.
+        @param name The name of the PointState you want to set to the new PointState.
+        @return A new PointState object containing the given value and name. 
+        """
+
+        member = object.__new__(cls)
+        member._value_ = value
+        member.__name__ = name
+        return member
+
+    def __int__(self):
+        """
+        @brief Used to cast the object to an int.
+        @details Returns the value of the class when int() is called.
+        @return Integer containing the value of the object.
+        """
+
+        return self.value
+
+
+class MapInterface():
```

Comment:
It looks like both `KeyValueMap` and `ListMap` have implemented getters for the `top_left_point` and `bot_right_point` why aren't they part of the interface?

Reply:
Same goes for other helper functions such as `width` and `height` etc